airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -128,18 +128,28 @@ def _create_session_with_assume_role(self, session_kwargs: Dict[str, Any]) -> bo
             )
         session = botocore.session.get_session()
         session._credentials = credentials
-        region_name = self.basic_session.region_name
-        session.set_config_variable("region", region_name)
+        if self.basic_session is not None:
+            region_name = self.basic_session.region_name
+            session.set_config_variable("region", region_name)
```

Comment:
This suggests to me that `basic_session` _can’t_ actually be None (otherwise current code would crash), so we should instead raise an exception for that case.

Reply:
Yeah. looked at the code and the basic_session **SHOULD** be there.